### PR TITLE
see phab T4394

### PR DESCRIPTION
**Pull request by examknow**

### CODE REVIEW

**Comment by examknow** [comment]
Travis CI build is just a sonar cloud error. It does not mean anything.

File: extension.json [hunk 1]
```diff
@@ -160,7 +160,10 @@
 			"description": "Boolean. Whether to implement front end logic for marking wikis as private.",
 			"public": true,
 			"value": false
-		}
+		},
```
Comment (paladox) [inline extension.json]: Needs to be formatted correctly please.

File: includes/CreateWiki/SpecialCreateWiki.php [hunk 2]
```diff
@@ -5,7 +5,11 @@ public function __construct() {
 	}
 
 	protected function getFormFields() {
-		global $wgCreateWikiUseCategories, $wgCreateWikiCategories, $wgCreateWikiUsePrivateWikis, $wgCreateWikiCDBDirectory;
+		$config = MediaWikiServices::getInstance()->getConfigFactory()->makeConfig( 'createwiki' );
+		$UseCategories = $config->get( 'UseCategories' );
```
Comment (paladox) [inline includes/CreateWiki/SpecialCreateWiki.php]: Convention for variable names is lowercase to upper case, so for example $useCategories. (And for the rest)

File: includes/RequestWiki/SpecialRequestWiki.php [hunk 3]
```diff
@@ -144,6 +144,8 @@ public function onSubmit( array $formData ) {
 
 		$request = $this->getRequest();
 
+
```
Comment (paladox) [inline includes/RequestWiki/SpecialRequestWiki.php]: Remove double new lines please.

File: includes/RequestWiki/SpecialRequestWiki.php [hunk 4]
```diff
@@ -160,7 +162,15 @@ public function onSubmit( array $formData ) {
...
-
+		
+		$dbr = wfGetDB( DB_REPLICA );
+		$res = $dbr->select( 'cw_requests', 'cw_dbname', __METHOD__ );
+		
+		if ( $dbname == $res ) {
+			$out->addHTML( '<div class="errorbox">' .  $this->msg( 'createwiki-error-wikitaken' )->escaped() . '</div>' );
+			wfDebugLog( 'CreateWiki', 'Wiki name already in use. Requested: ' . $dbname );
+			return false;
+					}
```
Comment (paladox) [inline includes/RequestWiki/SpecialRequestWiki.php]: Needs to be formatted correctly. And the rest.

**Comment by JohnFLewis** [comment]
I don't like the use of assigning temporary variables for single usage, it's much cleaner and easier to read using the $config->get call instead of assigning it to a variable.

Plus you're opening up unnecessary DB connections in new changes (which shouldn't be happening really in what should be a noop change).

Then you're comparing strings ($dbame) to database objects ($res) which will always return false.

**Comment by RhinosF1** [comment]
has errors, no one working on

**Pull request closed without merging**